modules/manifold/src/components/performance-comparison-control-container.js

```diff
@@ -71,116 +90,206 @@ class PerfroamnceComparisonControlContainer extends PureComponent {
+    return (
+      <InputButtons>
+        <button
+          disabled={isModelsComparisonLoading || isManualSegmentation}
+          onClick={() => this._onUpdateNClusters({isInc: false})}
+        >
+          -
+        </button>
+        <button
+          disabled={isModelsComparisonLoading || isManualSegmentation}
+          onClick={() => this._onUpdateNClusters({isInc: true})}
+        >
+          +
+        </button>
+      </InputButtons>
+    );
+  };
+
+  _onUpdateMetric = ({value}) => {
+    this.props.updateMetric(value[0]);
+  };
+
+  _onUpdateSegmentationMethod = ({value}) => {
+    this.props.updateSegmentationMethod(value[0].id);
   };
 
-  _onUpdateMetric = metric => {
-    const {modelComparisonParams, hasBackend} = this.props;
-    this.props.updateMetric(metric);
+  _onUpdateBaseCols = ({value}) => {
+    const colIds = value.map(col => col.tableFieldIndex - 1).sort();
+    this.props.updateBaseCols(colIds);
+  };
+
+  _onUpdateNClusters = ({isInc}) => {
+    this.props.updateNClusters(isInc ? 'INC' : 'DEC');
+  };
+
+  _updateSegmentGroups = segmentGroups => {
+    const {hasBackend, featureDistributionParams} = this.props;
+
+    // TODO constraints each group should have at least one segment
+    this.props.updateSegmentGroups(segmentGroups);
+    // TODO do we still need the hasBackend logic?
     if (hasBackend) {
-      this.props.fetchModels({
-        ...modelComparisonParams,
-        metric,
+      this.props.fetchFeatures({
+        ...featureDistributionParams,
+        segmentGroups,
       });
     }
   };
 
-  _onUpdateSegmentationMethod = e => {
-    this.props.updateSegmentationMethod(e.target.value);
-  };
-
   render() {
     // todo: support comparison view for multi-class cases
     const {
       className,
       width,
       flexDirection,
-      modelComparisonParams,
       isModelsComparisonLoading,
+      modelsMeta: {nClasses},
+      columnDefs,
+      metric,
       isManualSegmentation,
-      modelsMeta: {nClasses = 1} = {},
+      baseCols,
+      nClusters,
+      segmentFilters,
+      segmentIds,
+      segmentGroups,
     } = this.props;
-    const {nClusters} = modelComparisonParams;
     const isHorizontal = flexDirection === 'row';
+    const modelType =
+      nClasses === 1
+        ? MODEL_TYPE.REGRESSION
+        : nClasses === 2
+        ? MODEL_TYPE.BIN_CLASS
+        : MODEL_TYPE.MULT_CLASS;
+
+    // reset SegmentFiltersControl to default state whenever baseCols change
+    // https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key
+    const segmentFilterControlKey = baseCols
+      .slice()
+      .sort()
+      .join(',');
+
     return (
       <div className={className}>
         <StyledControl
-          name="Segmentation Method"
+          name="Performance Metric"
+          stackDirection={flexDirection}
+          isHidden={isHorizontal && width < WIDTH_LADDER[2]}
```

Comment:
Not in this PR, but let's add an issue to pull these `width < WIDTH_LADDER[*]` logic into a documented function.
I cannot follow how the indices map to the controls and what determines the order.

Reply:
Will do